qiskit/synthesis/linear/linear_depth_lnn.py

```diff
@@ -222,8 +222,8 @@ def _optimize_cx_circ_depth_5n_line(mat):
 
     # According to [1] the synthesis is done on the inverse matrix
     # so the matrix mat is inverted at this step
-    mat_inv = mat.copy()
-    mat_cpy = calc_inverse_matrix(mat_inv)
+    mat_inv = mat.copy().astype(bool)
```

Comment:
This is doing 2 copies and you only need one. You can do that with:

```suggestion
    mat_inv = np.asarray(mat, copy=True, dtype=bool)
```